cpp/oneapi/dal/algo/logistic_regression/backend/gpu/train_kernel_dense_batch_dpc.cpp

```diff
@@ -88,6 +88,8 @@ static train_result<Task> call_dal_kernel(const context_gpu& ctx,
 
     auto [train_event, iter_num] = opt_impl->minimize(queue, loss_func, x_suf, { fill_event });
 
+    train_event.wait_and_throw();
```

Comment:
Please, add TODO for investigate the reason of sporadic failures(probably it affects other algorithms as well)